src/librbd/io/ImageRequestWQ.cc

```diff
@@ -535,9 +554,15 @@ void ImageRequestWQ<I>::process(ImageRequest<I> *req) {
                  << "req=" << req << dendl;
 
   req->send();
+  bool is_write = req->is_write_op();
+  if (is_write && m_image_ctx.max_write_iops) {
+    write_qos_status.set_prev_limit(ceph_clock_now());    
```

Comment:
Once multiple thread pool threads are enabled, this won't be thread safe. If this is moved to the ```_void_dequeue``` method, it should be safe.